### PR TITLE
lock around audio-client quit flag to avoid shutdown race

### PR DESCRIPTION
**Pull request by sethalves**
- possible fix for crash-on-shutdown 2159

### CODE REVIEW

**Comment by sethalves** [comment]
to test:
- no functional change

**Comment by sethalves** [comment]
https://highfidelity.fogbugz.com/f/cases/2159/audio-client-shutdown-crash

File: libraries/audio-client/src/AudioClient.cpp [hunk 1]
```diff
@@ -85,18 +85,26 @@ class CheckDevicesThread : public QThread {
     }
 
     void beforeAboutToQuit() {
+        Lock lock(_checkDevicesMutex);
         _quit = true;
     }
 
     void run() override {
-        while (!_quit) {
+        for (;;) {
```
Comment (ZappoMan) [inline libraries/audio-client/src/AudioClient.cpp]: I feel like the coding standard has something to say about this...

**Comment by hifi-gustavo** [comment]
Build console output for commit: ae17135a075bec90f540d322718465e43bf766c1 available at: https://gist.github.com/cc89c4bdfee1313c0d25d65dfb14b225<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8972-ae17135.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8972-ae17135.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 137730dc50c66f3045a25d9ab874126ff479b2c0 available at: https://gist.github.com/cc89c4bdfee1313c0d25d65dfb14b225<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8972-137730d.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8972-137730d.dmg

**Comment by kevinmthomasse** [comment]
testing

**Comment by kevinmthomasse** [comment]
## QA Test
**QA Tester**: KevinMThomas
**QA Pass**: **QA Pass** FAILURE
**Additional Test Results + Observations**:  Failure as Audio Mute UI is non-existent.  Screen-Shot enclosed.
capture1

**Comment by sethalves** [comment]
The audio-mute buttons were removed (on purpose) in master -- unrelated to this PR.

**Comment by ZappoMan** [comment]
@kevinmthomasse - the audio mute UI is removed in master... it's now only visible as part of the HUD icon bar...

**Comment by kevinmthomasse** [comment]
@sethalves and @ZappoMan my error.  I update my report to PASS due to this understanding.  Are there additional testing notes that are relevant before merging?

**Pull request merged**